app/js/openmrs.js

```diff
@@ -0,0 +1,119 @@
+angular.module('openmrs.resources', [ 'ngResource' ])
+	.factory('openmrsApi', function($resource) {
+
+			var getOpenmrsContextPath = function() {
+				var pathname = window.location.pathname; 
+				return pathname.substring(0, pathname.indexOf("/owa/"));
+			}
+
+			var openmrsApi = {
+
+				defaultConfig : {
+					uuid : '@uuid',
+					mode : '@mode'
+				},
+
+				extraMethods : {
+
+				},
+
+				add : function(config) {
+					var params, url;
+
+					// If the add() function is called with a
+					// String, create the default configuration.
+					if (angular.isString(config)) {
+						var configObj = {
+							resource : config,
+							url : '/' + config
+						};
+
+						config = configObj;
+					}
+
+					// If the url follows the expected pattern, we can set cool
+					// defaults
+					if (!config.unnatural) {
+						var orig = angular.copy(openmrsApi.defaultConfig);
+						params = angular.extend(orig, config.params);
+						url = getOpenmrsContextPath() + '/ws/rest/v1' + config.url + '/:uuid?:mode';
+
+						// otherwise we have to declare the entire
+						// configuration.
+					} else {
+						params = config.params;
+						url = getOpenmrsContextPath() + '/ws/rest/v1' + config.url + '/:uuid?:mode';
+					}
+
+					// If we supply a method configuration, use that instead of
+					// the default extra.
+					var methods = config.methods || openmrsApi.extraMethods;
+
+					openmrsApi[config.resource] = $resource(url, params, methods);
+				}
+			};
+
+			return openmrsApi;
+		})
+
+	.provider('openmrsRest', {
```

Comment:
Could you please follow https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#style-y024?